### PR TITLE
feat(consortium-search): Implement consolidated items/holdings search

### PR DESCRIPTION
**Pull request by viacheslavkol**
### Purpose
Implement consolidated items/holdings search

### Approach
- Implement endpoints by id
- Implement batch endpoints by ids

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [x] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [ ] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MSEARCH-759](https://folio-org.atlassian.net/browse/MSEARCH-759)

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -179,6 +179,54 @@
...
             "user-tenants.collection.get"
           ]
+        },
+        {
+          "methods": [
+            "GET"
+          ],
+          "pathPattern": "/search/consortium/holding/{id}",
+          "permissionsRequired": [
+            "consortium-search.items.collection.get"
```
Comment (psmagin) [inline descriptors/ModuleDescriptor-template.json:189]:
```suggestion
            "consortium-search.holdings.item.get"
```

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -179,6 +179,54 @@
...
+            "user-tenants.collection.get"
+          ]
+        },
+        {
+          "methods": [
+            "GET"
+          ],
+          "pathPattern": "/search/consortium/item/{id}",
+          "permissionsRequired": [
+            "consortium-search.items.collection.get"
```
Comment (psmagin) [inline descriptors/ModuleDescriptor-template.json:201]:
```suggestion
            "consortium-search.items.item.get"
```

**Review by psmagin — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/search/controller/SearchConsortiumController.java [hunk 3]
```diff
@@ -82,11 +93,68 @@ public ResponseEntity<ConsortiumLocationCollection> getConsortiumLocations(Strin
...
+        .ok(new ConsortiumItemCollection());
+    }
+
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var itemIds = batchIdsDto.getIds().stream().map(UUID::toString).collect(Collectors.toSet());
+    var query = batchIdsDto.getIds().stream()
+      .map(UUID::toString)
+      .map("items.id = %s"::formatted)
+      .collect(Collectors.joining(" or "));
+    var searchRequest = CqlSearchRequest.of(Instance.class, tenant, query, itemIds.size(), 0, true, false, true);
```
Comment (mukhiddin-yusuf) [inline src/main/java/org/folio/search/controller/SearchConsortiumController.java:146]: looks like lines 142-146 could be extracted to some method and reused also in batch holdings endpoint for building a query and request

File: src/main/java/org/folio/search/converter/ConsortiumHoldingMapper.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.folio.search.converter;
+
+import org.folio.search.domain.dto.ConsortiumHolding;
+import org.folio.search.domain.dto.Holding;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (mukhiddin-yusuf) [inline src/main/java/org/folio/search/converter/ConsortiumHoldingMapper.java:7]: is it needed to have this as a bean? seems like we can just have it as a utility class (and also ConsortiumItemMapper)

**Review by mukhiddin-yusuf — APPROVED** [review]

File: src/main/java/org/folio/search/controller/SearchConsortiumController.java [hunk 5]
```diff
@@ -82,11 +97,72 @@ public ResponseEntity<ConsortiumLocationCollection> getConsortiumLocations(Strin
...
+    return ResponseEntity.ok(result);
+  }
+
+  private String verifyAndGetTenant(String tenantHeader) {
     var centralTenant = consortiumTenantService.getCentralTenant(tenantHeader);
     if (centralTenant.isEmpty() || !centralTenant.get().equals(tenantHeader)) {
       throw new RequestValidationException(REQUEST_NOT_ALLOWED_MSG, XOkapiHeaders.TENANT, tenantHeader);
     }
+    return centralTenant.get();
+  }
```
Comment (viacheslavpoliakov) [inline src/main/java/org/folio/search/controller/SearchConsortiumController.java:153]: I don't like methods like doSomeAndDoSomeAnd...
May be better way to do something like next
```
private String getVerifiedTenant(String tenantHeader) {
    verifyTenant(tenantHeader)
    return consortiumTenantService.getCentralTenant(tenantHeader).get();
}

private void verifyTenant(String tenantHeader) {
    var centralTenant = consortiumTenantService.getCentralTenant(tenantHeader);
    if (centralTenant.isEmpty() || !centralTenant.get().equals(tenantHeader)) {
      throw new RequestValidationException(REQUEST_NOT_ALLOWED_MSG, XOkapiHeaders.TENANT, tenantHeader);
    }
}
```

**Review by viacheslavpoliakov — APPROVED** [review]

**Pull request merged**